smtpd/mail.lmtp.c

```diff
@@ -209,17 +209,30 @@ lmtp_connect(const char *destination)
 }
 
 static void
-lmtp_engine(FILE *conn, struct session *session)
+lmtp_engine(int fd_read, struct session *session)
 {
+	int fd_write = 0;
+	FILE *file_read = 0;
+	FILE *file_write = 0;
 	char *line = NULL;
 	size_t linesize = 0;
 	ssize_t linelen;
 	enum phase phase = PHASE_BANNER;
 
+	if ((fd_write = dup(fd_read)) == -1)
+		err(EX_TEMPFAIL, "dup");
+
+	if (!(file_read = fdopen(fd_read, "r")))
+		err(EX_TEMPFAIL, "fdopen");
+
+	if (!(file_write = fdopen(fd_write, "w")))
```

Comment:
same

Reply:
Fixed in https://github.com/OpenSMTPD/OpenSMTPD/pull/999/commits/0d9aebb4197cf99020afed950fd9f43afe38427e